stackit/internal/services/serviceaccount/serviceaccount_acc_test.go

```diff
@@ -0,0 +1,159 @@
+	"name02":     "sa-test-02",
+}
+
+func inputServiceAccountResourceConfig(name string) string {
+	return fmt.Sprintf(`
+				%s
+			
+				resource "stackit_service_account" "sa" {
+					project_id = "%s"
+					name = "%s"
+				}
+				`,
+		testutil.ServiceAccountProviderConfig(),
+		serviceAccountResource["project_id"],
+		name,
+	)
+}
+
+func inputServiceAccountDataSourceConfig() string {
+	return fmt.Sprintf(`
+					%s
+
+					data "stackit_service_account" "sa" {
+						project_id  = stackit_service_account.sa.project_id
+						email = stackit_service_account.sa.email
+					}
+					`,
+		inputServiceAccountResourceConfig(serviceAccountResource["name01"]),
+	)
+}
+
+func TestServiceAccount(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: testutil.TestAccProtoV6ProviderFactories,
+		CheckDestroy:             testAccCheckServiceAccountDestroy,
+		Steps: []resource.TestStep{
+			// Creation
+			{
+				Config: inputServiceAccountResourceConfig(serviceAccountResource["name01"]),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("stackit_service_account.sa", "project_id", serviceAccountResource["project_id"]),
+					resource.TestCheckResourceAttr("stackit_service_account.sa", "name", serviceAccountResource["name01"]),
+					resource.TestCheckResourceAttrSet("stackit_service_account.sa", "email"),
+				),
+			},
+			// Update
+			{
+				Config: inputServiceAccountResourceConfig(serviceAccountResource["name02"]),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("stackit_service_account.sa", "project_id", serviceAccountResource["project_id"]),
+					resource.TestCheckResourceAttr("stackit_service_account.sa", "name", serviceAccountResource["name02"]),
+					resource.TestCheckResourceAttrSet("stackit_service_account.sa", "email"),
+				),
+			},
+			// Data source
+			{
+				Config: inputServiceAccountDataSourceConfig(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Instance
+					resource.TestCheckResourceAttr("data.stackit_service_account.sa", "project_id", serviceAccountResource["project_id"]),
+					resource.TestCheckResourceAttrPair(
+						"stackit_service_account.sa", "project_id",
+						"data.stackit_service_account.sa", "project_id",
+					),
+					resource.TestCheckResourceAttrPair(
+						"stackit_service_account.sa", "name",
+						"data.stackit_service_account.sa", "name",
+					),
+					resource.TestCheckResourceAttrPair(
+						"stackit_service_account.sa", "email",
+						"data.stackit_service_account.sa", "email",
+					),
+				),
+			},
+			// Import
+			{
+				ResourceName: "stackit_service_account.sa",
+				ImportStateIdFunc: func(s *terraform.State) (string, error) {
+					r, ok := s.RootModule().Resources["stackit_service_account.sa"]
+					if !ok {
+						return "", fmt.Errorf("couldn't find resource stackit_service_account.sa")
+					}
+					email := strings.Split(r.Primary.ID, ",")[1]
```

Comment:
Only a small nitpick but can be changed to this
```suggestion
					email, ok := r.Primary.Attributes["email"]
					if !ok {
						return "", fmt.Errorf("couldn't find attribute email")
					}
```